include/revng/Model/Register.h

```diff
@@ -936,6 +936,14 @@ constexpr inline model::PrimitiveTypeKind::Values primitiveKind(Values V) {
   }
 }
 
+constexpr inline bool isGeneralPurpose(Values V) {
+  return primitiveKind(V) == model::PrimitiveTypeKind::PointerOrNumber;
+}
+
+constexpr inline bool isVector(Values V) {
+  return primitiveKind(V) == model::PrimitiveTypeKind::Float;
+}
```

Comment:
You're introducing two new concepts "GeneralPurpose" and "Vector" which are never defined out of `librevngABI`, I'd make these methods private and reconsider these for `Register.h` once we get a more definitive stance over `Float` vs ` Vector`.

Reply:
You have a point, I'll localize these.

Reply:
Done.